### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by jskeet**

### CODE REVIEW

**Comment by jskeet** [comment]
(Once this is in, I'll rebuild and push the docs.)

**Comment by jskeet** [comment]
I can't see why AppVeyor thinks this is non-mergeable, btw...

**Comment by jskeet** [comment]
(AppVeyor now running after I rebased post-DLP merge...)

**Comment by jskeet** [comment]
Added another commit to fix the AppVeyor issue - PTAL.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+namespace Google.Cloud.Spanner.Data.Snippets
+{
+    public class SpannerConnectionSnippets
+    {
+        [Fact]
+        public void CreateConnection()
+        {
+            // Snippet: #ctor(string, ITokenAccess)
+            string connectionString = $"Data Source=projects/my-project/instances/my-instance/databases/my-db";
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs]: $ Not needed.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Snippets
+{
+    public class SpannerConnectionSnippets
+    {
+        [Fact]
+        public void CreateConnection()
+        {
+            // Snippet: #ctor(string, ITokenAccess)
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs]: Just to check that #ctor is correct?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs (hunk 1 above)
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs]: Doh, thanks. (Previously had variables.)

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs (hunk 2 above)
Comment (jskeet) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Snippets/SpannerConnectionSnippets.cs]: Yup. Consulted the YAML file after just using SpannerConnection(...) didn't work. Would really like to get the v2 of snippets using Roslyn work going...

**Comment by jskeet** [comment]
Will merge if the *current* AppVeyor build (with the $) passes.

**Pull request merged**